File: backend/src/main/java/com/ro/petrol_pump_ai/entity/Attendance.java
package com.ro.petrol_pump_ai.entity;

import lombok.*;
import jakarta.persistence.*;
import java.time.LocalDate;
import java.time.LocalTime;
import java.time.LocalDateTime;

@Entity
@Table(name = "attendance")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Attendance {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne
    @JoinColumn(name = "employee_id", nullable = false)
    private Employee employee;
    
    @Column(nullable = false)
    private LocalDate attendanceDate;
    
    private LocalTime checkInTime;
    private LocalTime checkOutTime;
    
    @Column(nullable = false)
    private String status;  // Present, Absent, Leave, Half-day
    
    private String remarks;
    
    private LocalDateTime createdAt;
    
    @PrePersist
    public void onCreate() {
        this.createdAt = LocalDateTime.now();
    }
}

File: frontend/src/types/attendance.ts
export interface Attendance {
    id: number;
    employeeId: number;
    employeeName: string;
    attendanceDate: string;
    checkInTime: string | null;
    checkOutTime: string | null;
    status: "Present" | "Absent" | "Late" | "Leave";
    durationHours: number | null;
    notes: string;
  }
  
  export interface AttendanceRequest {
    employeeId: number;
    attendanceDate: string;
    checkInTime: string | null;
    checkOutTime: string | null;
    status: string;
    notes: string;
  }
  